### PR TITLE
[WIP] Experimental watch events trace

### PR DESCRIPTION
**Pull request by mborsz**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This implements experimental watch trace to help understanding why some watch has been closed. Without this, the only info we get is that "the client was too slow" which doesn't help that much unfortunately.

This is WIP change. I'm going to run few manual runs with it and see if it is valuable. Then, we may consider merging this either as is or behind some flag.

**Which issue(s) this PR fixes**:

Ref https://github.com/kubernetes/kubernetes/issues/97798
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by mborsz** [comment]
FTR: While testing this logging I noticed interesting series of watch closures.

```
I0118 13:29:00.468775      12 cacher.go:1261] [0xc07f90da00] Forcing *discovery.EndpointSlice watcher close due to unresponsiveness: key: "/endpointslices", labels: "!service.kubernetes.io/headless,!service.kubernetes.io/service-proxy-name", fields: "". len(c.input) = 10, len(c.result) = 0

I0118 13:29:00.505470      12 watch.go:269] [0xc07f90da00] Watch "/apis/discovery.k8s.io/v1beta1/endpointslices?allowWatchBookmarks=true&labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&resourceVersion=979875&timeout=7m3s&timeoutSeconds=423&watch=true" from "kube-proxy/v1.21.0 (linux/amd64) kubernetes/91848e1"(
"10.40.18.123:56228") unexpectedly stopped. Last event trace: [
event(start: 2021-01-18 13:29:00.24508617 +0000 UTC m=+1429.657692912, encode latency: 39.387µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245128515 +0000 UTC m=+1429.657735247, encode latency: 40.574µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245171446 +0000 UTC m=+1429.657778185, encode latency: 37.613µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245210815 +0000 UTC m=+1429.657817539, encode latency: 45.761µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245258661 +0000 UTC m=+1429.657865385, encode latency: 136.771µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245398448 +0000 UTC m=+1429.658005184, encode latency: 50.831µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245451753 +0000 UTC m=+1429.658058476, encode latency: 108.016µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245571574 +0000 UTC m=+1429.658178305, encode latency: 148.762µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245729193 +0000 UTC m=+1429.658335931, encode latency: 60.042µs, flush latency: 0s)
event(start: 2021-01-18 13:29:00.245791496 +0000 UTC m=+1429.658398222, encode latency: 119.272µs, flush latency: 0s)]
```

```
I0118 13:28:59.955829      12 cacher.go:1261] [0xc2c5fa0200] Forcing *discovery.EndpointSlice watcher close due to unresponsiveness: key: "/endpointslices", labels: "!service.kubernetes.io/headless,!service.kubernetes.io/service-proxy-name", fields: "". len(c.input) = 10, len(c.result) = 10
I0118 13:29:00.026383      12 watch.go:269] [0xc2c5fa0200] Watch "/apis/discovery.k8s.io/v1beta1/endpointslices?allowWatchBookmarks=true&labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&resourceVersion=802889&timeout=5m21s&timeoutSeconds=321&watch=true" from "kube-proxy/v1.21.0 (linux/amd64) kubernetes/91848e1"("10.40.10.179:45900") unexpectedly stopped. Last event trace: [
event(start: 2021-01-18 13:28:59.965783075 +0000 UTC m=+1429.378389804, encode latency: 220.329µs, flush latency: 0s)
event(start: 2021-01-18 13:28:59.966009633 +0000 UTC m=+1429.378616365, encode latency: 18.131µs, flush latency: 0s)
event(start: 2021-01-18 13:28:59.966029639 +0000 UTC m=+1429.378636363, encode latency: 74.453µs, flush latency: 0s)
event(start: 2021-01-18 13:28:59.966106458 +0000 UTC m=+1429.378713179, encode latency: 6.111µs, flush latency: 0s) 
event(start: 2021-01-18 13:28:59.966113932 +0000 UTC m=+1429.378720660, encode latency: 41.699µs, flush latency: 0s)
event(start: 2021-01-18 13:28:59.96615769 +0000 UTC m=+1429.378764417, encode latency: 62.567µs, flush latency: 0s)
event(start: 2021-01-18 13:28:59.966222624 +0000 UTC m=+1429.378829352, encode latency: 112.239µs, flush latency: 0s)
event(start: 2021-01-18 13:28:59.966338806 +0000 UTC m=+1429.378945533, encode latency: 12.913µs, flush latency: 0s)
event(start: 2021-01-18 13:28:59.966353467 +0000 UTC m=+1429.378960188, encode latency: 3.642µs, flush latency: 0s)
event(start: 2021-01-18 13:28:59.966358792 +0000 UTC m=+1429.378965571, encode latency: 3.308µs, flush latency: 0s)]
I0118 13:29:00.026698      12 httplog.go:89] "HTTP" verb="GET" URI="/apis/discovery.k8s.io/v1beta1/endpointslices?allowWatchBookmarks=true&labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&resourceVersion=802889&timeout=5m21s&timeoutSeconds=321&watch=true" latency="3m23.010148056s" userAgent="kube-proxy/v1.21.0 (linux/amd64) kubernetes/91848e1" srcIP="10.40.10.179:45900" resp=0
```

**Comment by mborsz** [comment]
Also seems to give some information:

```
I0118 13:19:28.398900      12 cacher.go:1261] [0xc23d771000] Forcing *coordination.Lease watcher close due to unresponsiveness: key: "/leases", labels: "", fields: "". len(c.input) = 10, len(c.result) = 10
I0118 13:19:28.833332      12 cacher.go:1261] [0xc2789abc80] Forcing *coordination.Lease watcher close due to unresponsiveness: key: "/leases", labels: "", fields: "". len(c.input) = 10, len(c.result) = 2
I0118 13:19:28.992948      12 cacher.go:1261] [0xc14a82fd00] Forcing *coordination.Lease watcher close due to unresponsiveness: key: "/leases", labels: "", fields: "". len(c.input) = 10, len(c.result) = 5
I0118 13:19:28.998080      12 cacher.go:1261] [0xc16aff8b00] Forcing *coordination.Lease watcher close due to unresponsiveness: key: "/leases", labels: "", fields: "". len(c.input) = 10, len(c.result) = 0
I0118 13:19:28.999333      12 cacher.go:1261] [0xc22c62f200] Forcing *coordination.Lease watcher close due to unresponsiveness: key: "/leases", labels: "", fields: "". len(c.input) = 10, len(c.result) = 4
```

My interpretation is that in fact a goroutine processing c.input and writing to c.result is a bottleneck here (a c.input is full and c.result is empty)

**Comment by mborsz** [comment]
From the next run (with 25 buffer size):

```
I0119 09:00:01.237414      11 cacher.go:1261] [0xc1b629e880] Forcing *discovery.EndpointSlice watcher close due to unresponsiveness: key: "/endpointslices", labels: "!service.kubernetes.io/headless,!service.kubernetes.io/service-proxy-name", fields: "". len(c.input) = 10, len(c.result) = 10
I0119 09:00:01.318799      11 watch.go:274] [0xc1b629e880] Watch "/apis/discovery.k8s.io/v1beta1/endpointslices?allowWatchBookmarks=true&labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&resourceVersion=511174&timeout=5m39s&timeoutSeconds=339&watch=true" from "kube-proxy/v1.21.0 (linux/amd64) kubernetes/a62a26b"("10.40.14.91:50218") unexpectedly stopped. Last event trace: [
event(start: 2021-01-19 09:00:00.088209241 +0000 UTC m=+1094.905469606, encode latency: 10.123µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.104807484 +0000 UTC m=+1094.922068109, encode latency: 10.202µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.118243831 +0000 UTC m=+1094.935504168, encode latency: 7.802µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.129232333 +0000 UTC m=+1094.946492712, encode latency: 8.465µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.140154664 +0000 UTC m=+1094.957415001, encode latency: 10.337µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.174505984 +0000 UTC m=+1094.991766323, encode latency: 8.51µs, flush latency: 0s) 
event(start: 2021-01-19 09:00:00.196868382 +0000 UTC m=+1095.014128750, encode latency: 9.833µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.213519565 +0000 UTC m=+1095.030780090, encode latency: 9.087µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.262256299 +0000 UTC m=+1095.079516998, encode latency: 9.608µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.269264882 +0000 UTC m=+1095.086525290, encode latency: 11.889µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.279088882 +0000 UTC m=+1095.096349315, encode latency: 19.775µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.313594005 +0000 UTC m=+1095.130854348, encode latency: 10.559µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.338180377 +0000 UTC m=+1095.155440712, encode latency: 9.892µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.343112216 +0000 UTC m=+1095.160372558, encode latency: 14.598µs, flush latency: 0s)
event(start: 2021-01-19 09:00:00.369955429 +0000 UTC m=+1095.187215765, encode latency: 945.98293ms, flush latency: 0s)
event(start: 2021-01-19 09:00:01.315941703 +0000 UTC m=+1096.133202044, encode latency: 5.679µs, flush latency: 0s)
event(start: 2021-01-19 09:00:01.31594923 +0000 UTC m=+1096.133209570, encode latency: 4.489µs, flush latency: 0s) 
event(start: 2021-01-19 09:00:01.315955439 +0000 UTC m=+1096.133215773, encode latency: 4.459µs, flush latency: 0s)
event(start: 2021-01-19 09:00:01.315962462 +0000 UTC m=+1096.133222797, encode latency: 1.344658ms, flush latency: 0s)
event(start: 2021-01-19 09:00:01.317310578 +0000 UTC m=+1096.134570914, encode latency: 8.164µs, flush latency: 0s) 
event(start: 2021-01-19 09:00:01.317320239 +0000 UTC m=+1096.134580575, encode latency: 3.805µs, flush latency: 0s)
event(start: 2021-01-19 09:00:01.317326232 +0000 UTC m=+1096.134586571, encode latency: 4.335µs, flush latency: 0s)
event(start: 2021-01-19 09:00:01.317332146 +0000 UTC m=+1096.134592480, encode latency: 818.207µs, flush latency: 0s)
event(start: 2021-01-19 09:00:01.318153618 +0000 UTC m=+1096.135413939, encode latency: 9.021µs, flush latency: 0s) 
event(start: 2021-01-19 09:00:01.31816428 +0000 UTC m=+1096.135424616, encode latency: 4.167µs, flush latency: 0s)]
I0119 09:00:01.319102      11 httplog.go:89] "HTTP" verb="GET" URI="/apis/discovery.k8s.io/v1beta1/endpointslices?allowWatchBookmarks=true&labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&resourceVersion=511174&timeout=5m39s&timeoutSeconds=339&watch=true" latency="4m15.14526611s" userAgent="kube-proxy/v1.21.0 (linux/amd64) kubernetes/a62a26b" srcIP="10.40.14.91:50218" resp=0
```

The line


```
event(start: 2021-01-19 09:00:00.369955429 +0000 UTC m=+1095.187215765, encode latency: 945.98293ms, flush latency: 0s)
```
Is interesting. The 945ms latency seems to be quite high.

**Comment by mborsz** [comment]
Another run:

```
I0119 12:42:00.150124      11 cacher.go:1261] [0xc0c31f1c00] Forcing *discovery.EndpointSlice watcher close due to unresponsiveness: key: "/endpointslices", labels: "!service.kubernetes.io/headless,!service.kubernetes.io/service-proxy-name", fields: "". len(c.input) = 10, len(c.result) = 10

I0119 12:42:00.392369      11 watch.go:284] [0xc0c31f1c00] Watch "/apis/discovery.k8s.io/v1beta1/endpointslices?allowWatchBookmarks=true&labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&resourceVersion=680552&timeout=6m9s&timeoutSeconds=369&watch=true" from "kube-proxy/v1.21.0 (linux/amd64) kubernetes/9bfa778"("10.40.1.32:50982") unexpectedly stopped. Last event trace: event(start: 2021-01-19 12:41:59.101871722 +0000 UTC m=+1187.253263956, latency: 3.031259ms [c: 8.736µs, e: 3.021595ms, f: 928ns])
event(start: 2021-01-19 12:41:59.121859974 +0000 UTC m=+1187.273251972, latency: 315.077µs [c: 3.924µs, e: 18.757µs, f: 292.396µs])
event(start: 2021-01-19 12:41:59.149412122 +0000 UTC m=+1187.300804118, latency: 904.866µs [c: 25.327µs, e: 7.252µs, f: 872.287µs])
event(start: 2021-01-19 12:41:59.164497118 +0000 UTC m=+1187.315889091, latency: 10.028µs [c: 5.459µs, e: 4.44µs, f: 129ns])
event(start: 2021-01-19 12:41:59.16450887 +0000 UTC m=+1187.315900844, latency: 855.481µs [c: 2.603µs, e: 2.551µs, f: 850.327µs])
event(start: 2021-01-19 12:41:59.184722314 +0000 UTC m=+1187.336114312, latency: 529.141µs [c: 24.517µs, e: 11.852µs, f: 492.772µs])
event(start: 2021-01-19 12:41:59.194360538 +0000 UTC m=+1187.345752523, latency: 602.32µs [c: 4.307µs, e: 7.786µs, f: 590.227µs])
event(start: 2021-01-19 12:41:59.231395134 +0000 UTC m=+1187.382787115, latency: 14.488µs [c: 7.207µs, e: 7.152µs, f: 129ns])
event(start: 2021-01-19 12:41:59.23141149 +0000 UTC m=+1187.382803471, latency: 1.938921ms [c: 23.01µs, e: 1.914981ms, f: 930ns])
event(start: 2021-01-19 12:41:59.261819928 +0000 UTC m=+1187.413212022, latency: 946.403µs [c: 5.073µs, e: 58.318µs, f: 883.012µs])
event(start: 2021-01-19 12:41:59.281957521 +0000 UTC m=+1187.433349499, latency: 1.348815ms [c: 4.89µs, e: 9.013µs, f: 1.334912ms])
event(start: 2021-01-19 12:41:59.294060516 +0000 UTC m=+1187.445452493, latency: 1.156909ms [c: 4.597µs, e: 4.391µs, f: 1.147921ms])
event(start: 2021-01-19 12:41:59.307951941 +0000 UTC m=+1187.459343923, latency: 6.89026ms [c: 41.574µs, e: 6.848227ms, f: 459ns])
event(start: 2021-01-19 12:41:59.314847758 +0000 UTC m=+1187.466239735, latency: 38.921137ms [c: 33.562µs, e: 38.885502ms, f: 2.073µs])
event(start: 2021-01-19 12:41:59.416230156 +0000 UTC m=+1187.567622296, latency: 965.700385ms [c: 965.692831ms, e: 7.219µs, f: 335ns])
event(start: 2021-01-19 12:42:00.381934622 +0000 UTC m=+1188.533326603, latency: 6.621µs [c: 4.407µs, e: 2.12µs, f: 94ns])
event(start: 2021-01-19 12:42:00.381942341 +0000 UTC m=+1188.533334321, latency: 1.250378ms [c: 2.176µs, e: 1.24744ms, f: 762ns])
event(start: 2021-01-19 12:42:00.38319745 +0000 UTC m=+1188.534589431, latency: 2.900296ms [c: 10.06µs, e: 2.889733ms, f: 503ns])
event(start: 2021-01-19 12:42:00.386101462 +0000 UTC m=+1188.537493439, latency: 9.849µs [c: 4.729µs, e: 5.037µs, f: 83ns])
event(start: 2021-01-19 12:42:00.386112529 +0000 UTC m=+1188.537504502, latency: 3.039968ms [c: 4.501µs, e: 3.035014ms, f: 453ns])
event(start: 2021-01-19 12:42:00.389155908 +0000 UTC m=+1188.540547897, latency: 8.437µs [c: 4.633µs, e: 3.705µs, f: 99ns])
event(start: 2021-01-19 12:42:00.389165942 +0000 UTC m=+1188.540557926, latency: 3.794µs [c: 2.068µs, e: 1.631µs, f: 95ns])
event(start: 2021-01-19 12:42:00.389171157 +0000 UTC m=+1188.540563131, latency: 8.985µs [c: 2.306µs, e: 6.581µs, f: 98ns])
event(start: 2021-01-19 12:42:00.389181677 +0000 UTC m=+1188.540573660, latency: 4.376µs [c: 2.66µs, e: 1.62µs, f: 96ns])
event(start: 2021-01-19 12:42:00.38918719 +0000 UTC m=+1188.540579167, latency: 2.975307ms [c: 2.505µs, e: 1.725241ms, f: 1.247561ms])
I0119 12:42:00.392447      11 watch.go:285] [0xc0c31f1c00] Encoder: *versioning.codec, ResponseWriter: *http2.responseWriter
```

So, interestingly most of the time we spent on in-memory operations: 965.692831ms which suggests some CPU starvation or some issue with memory allocation.

File: staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go [hunk 1]
```diff
@@ -1258,7 +1258,7 @@ func (c *cacheWatcher) add(event *watchCacheEvent, timer *time.Timer) bool {
 		// This means that we couldn't send event to that watcher.
 		// Since we don't want to block on it infinitely,
 		// we simply terminate it.
-		klog.V(1).Infof("Forcing %v watcher close due to unresponsiveness: %v", c.objectType.String(), c.identifier)
+		klog.V(1).Infof("[%p] Forcing %v watcher close due to unresponsiveness: %v. len(c.input) = %v, len(c.result) = %v", c, c.objectType.String(), c.identifier, len(c.input), len(c.result))
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go:1261]: I think that this is universally good - so let's extract this one to a separate PR and have that merge.

[The other changes may potentially slightly affect performance, so I would like to first see their impact]

Comment (mborsz) [reply staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go:1261]: Please see https://github.com/kubernetes/kubernetes/pull/98208

**Comment by mborsz** [comment]
Another interesting case (from https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-golang-tip-k8s-e2-1-18/1352038062514769920 where I manually cherry-picked this change):

```
I0121 00:00:47.882677       1 cacher.go:1232] [0xc056329300] Forcing *core.Endpoints watcher close due to unresponsiveness: key: "/services/endpoints", labels: "!service.kubernetes.io/headless,!service.kubernetes.io/service-proxy-name", fields: "". len(c.input) = 10, len(c.result) = 10
I0121 00:00:48.791573       1 watch.go:293] [0xc056329300] Watch "/api/v1/endpoints?allowWatchBookmarks=true&labelSelector=%21service.kubernetes.io%2Fheadless%2C%21service.kubernetes.io%2Fservice-proxy-name&resourceVersion=137890&timeout=5m4s&timeoutSeconds=304&watch=true" from "kubemark/v1.18.3 (linux/amd64) kubernetes/16f3f46"("10.64.31.55:59232") unexpectedly stopped. Last event trace:
event(start: 2021-01-21 00:00:46.050026922 +0000 UTC m=+703.511406345, latency: 14.629693ms [ee: 13.853µs, c: 2.62µs, e: 14.612561ms, f: 659ns], size: 17616)
event(start: 2021-01-21 00:00:46.116683854 +0000 UTC m=+703.578063337, latency: 2.773632ms [ee: 11.394µs, c: 3.18µs, e: 2.758489ms, f: 569ns], size: 19244)
event(start: 2021-01-21 00:00:46.394925325 +0000 UTC m=+703.856304783, latency: 13.045746ms [ee: 10.85µs, c: 2.53µs, e: 13.0317ms, f: 666ns], size: 20724)
event(start: 2021-01-21 00:00:46.546244752 +0000 UTC m=+704.007624496, latency: 3.966415ms [ee: 14.094µs, c: 1.877µs, e: 3.949409ms, f: 1.035µs], size: 25016)
event(start: 2021-01-21 00:00:46.603277008 +0000 UTC m=+704.064656488, latency: 2.263575ms [ee: 17.793µs, c: 2.535µs, e: 2.242013ms, f: 1.234µs], size: 28124)
event(start: 2021-01-21 00:00:46.706988005 +0000 UTC m=+704.168367536, latency: 1.98632ms [ee: 19.373µs, c: 2.557µs, e: 1.963791ms, f: 599ns], size: 28272)
event(start: 2021-01-21 00:00:46.7752325 +0000 UTC m=+704.236611930, latency: 25.023671ms [ee: 18.089µs, c: 3.277µs, e: 25.001107ms, f: 1.198µs], size: 29012)
event(start: 2021-01-21 00:00:46.866231094 +0000 UTC m=+704.327610528, latency: 377.132655ms [ee: 19.234µs, c: 2.116µs, e: 377.110941ms, f: 364ns], size: 30492)
event(start: 2021-01-21 00:00:47.243369053 +0000 UTC m=+704.704748476, latency: 189.677µs [ee: 24.402µs, c: 2.678µs, e: 162.455µs, f: 142ns], size: 31676)
event(start: 2021-01-21 00:00:47.24356237 +0000 UTC m=+704.704941799, latency: 194.504785ms [ee: 8.377µs, c: 1.73µs, e: 194.494045ms, f: 633ns], size: 32712)
event(start: 2021-01-21 00:00:47.438073593 +0000 UTC m=+704.899453009, latency: 86.981771ms [ee: 20.428µs, c: 3.075µs, e: 86.957921ms, f: 347ns], size: 32689)
event(start: 2021-01-21 00:00:47.525060604 +0000 UTC m=+704.986440010, latency: 188.796µs [ee: 177.088µs, c: 3.563µs, e: 7.742µs, f: 403ns], size: 579)
event(start: 2021-01-21 00:00:47.525253422 +0000 UTC m=+704.986632846, latency: 93.064789ms [ee: 10.299µs, c: 1.583µs, e: 93.052497ms, f: 410ns], size: 33600)
event(start: 2021-01-21 00:00:47.618324554 +0000 UTC m=+705.079717285, latency: 381.975µs [ee: 18.06µs, c: 2.78µs, e: 360.918µs, f: 217ns], size: 33748)
event(start: 2021-01-21 00:00:47.618724677 +0000 UTC m=+705.080104098, latency: 442.737569ms [ee: 13.089µs, c: 2.418µs, e: 442.721525ms, f: 537ns], size: 34192)
event(start: 2021-01-21 00:00:48.062185645 +0000 UTC m=+705.523565078, latency: 195.572414ms [ee: 27.11µs, c: 3.352µs, e: 195.541472ms, f: 480ns], size: 34932)
event(start: 2021-01-21 00:00:48.257772206 +0000 UTC m=+705.719151629, latency: 5.887032ms [ee: 15.73µs, c: 3.013µs, e: 5.867995ms, f: 294ns], size: 35228)
event(start: 2021-01-21 00:00:48.263663875 +0000 UTC m=+705.725043322, latency: 16.795µs [ee: 8.357µs, c: 2.487µs, e: 5.85µs, f: 101ns], size: 606)
event(start: 2021-01-21 00:00:48.263682538 +0000 UTC m=+705.725061972, latency: 79.07272ms [ee: 11.11µs, c: 1.642µs, e: 79.059343ms, f: 625ns], size: 35524)
event(start: 2021-01-21 00:00:48.342761767 +0000 UTC m=+705.804141190, latency: 8.83915ms [ee: 27.392µs, c: 3.896µs, e: 8.807432ms, f: 430ns], size: 35672)
event(start: 2021-01-21 00:00:48.351607555 +0000 UTC m=+705.812986978, latency: 150.044122ms [ee: 22.202µs, c: 3.653µs, e: 150.017686ms, f: 581ns], size: 35820)
event(start: 2021-01-21 00:00:48.501658465 +0000 UTC m=+705.963037889, latency: 99.063936ms [ee: 19.15µs, c: 3.755µs, e: 99.040645ms, f: 386ns], size: 36116)
event(start: 2021-01-21 00:00:48.600727743 +0000 UTC m=+706.062107160, latency: 93.978314ms [ee: 18.162µs, c: 2.897µs, e: 93.956807ms, f: 448ns], size: 36264)
event(start: 2021-01-21 00:00:48.694710551 +0000 UTC m=+706.156089961, latency: 158.582µs [ee: 16.205µs, c: 2.38µs, e: 139.783µs, f: 214ns], size: 36412)
event(start: 2021-01-21 00:00:48.69487236 +0000 UTC m=+706.156251773, latency: 96.46667ms [ee: 9.526µs, c: 1.802µs, e: 96.453527ms, f: 1.815µs], size: 36708)
```

This is quite interesting, as this is a first trace where all events are just slow. Not sure what does it mean. What seems to be interesting is an object size: 36K for single object seems to be a lot, but may be quite standard for endpoints.
